internal/cloud/backend.go

```diff
@@ -411,9 +413,35 @@ func (b *Cloud) discover() (*url.URL, error) {
 		return nil, err
 	}
 
+	client := &http.Client{
```

Comment:
Trying to add retry logic here, but it's not retrying. Output from a local run:

```
$ TF_LOG=debug /Users/awinkler/.asdf/installs/golang/1.19.4/packages/bin/terraform apply
2023-01-10T19:15:11.644-0700 [INFO]  Terraform version: 1.4.0 dev
2023-01-10T19:15:11.644-0700 [DEBUG] using github.com/hashicorp/go-tfe v1.12.0
2023-01-10T19:15:11.644-0700 [DEBUG] using github.com/hashicorp/hcl/v2 v2.15.0
2023-01-10T19:15:11.644-0700 [DEBUG] using github.com/hashicorp/terraform-config-inspect v0.0.0-20210209133302-4fd17a0faac2
2023-01-10T19:15:11.644-0700 [DEBUG] using /Users/awinkler/dev/terraform-svchost (devel)
2023-01-10T19:15:11.644-0700 [DEBUG] using github.com/zclconf/go-cty v1.12.1
2023-01-10T19:15:11.644-0700 [INFO]  Go runtime version: go1.19.4
2023-01-10T19:15:11.644-0700 [INFO]  CLI args: []string{"/Users/awinkler/.asdf/installs/golang/1.19.4/packages/bin/terraform", "apply"}
2023-01-10T19:15:11.644-0700 [DEBUG] Attempting to open CLI config file: /Users/awinkler/.terraformrc
2023-01-10T19:15:11.645-0700 [INFO]  Loading CLI configuration from /Users/awinkler/.terraformrc
2023-01-10T19:15:11.645-0700 [INFO]  Loading CLI configuration from /Users/awinkler/.terraform.d/credentials.tfrc.json
2023-01-10T19:15:11.645-0700 [DEBUG] Explicit provider installation configuration is set
2023-01-10T19:15:11.645-0700 [INFO]  CLI command args: []string{"apply"}
2023-01-10T19:15:11.646-0700 [DEBUG] Before calling Discover...
2023-01-10T19:15:11.646-0700 [DEBUG] Service discovery for app.terraform.io at https://app.terraform.io/.well-known/terraform.json
╷
│ Error: Failed to request discovery document: Get "https://app.terraform.io/.well-known/terraform.json": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

Reply:
Looked at this with Brandon and saw that the client is being set in the svchost library, so rethinking the problem.